lib/KTextbox/__tests__/KTextbox.spec.js

```diff
@@ -0,0 +1,158 @@
+import { shallowMount, mount } from '@vue/test-utils';
+import KTextbox from '../index';
+
+describe('KTextbox component', () => {
+  it(`smoke test`, () => {
+    const wrapper = shallowMount(KTextbox);
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  describe('props', () => {
+    it(`a label should appear`, () => {
+      const wrapper = mount(KTextbox, {
+        propsData: {
+          label: 'test',
+        },
+      });
+      expect(wrapper.find('label').text()).toEqual('test');
+    });
+    it(`value of the text field should appear when passed in`, () => {
+      const initialValue = '1234paowiejfapwoeifjapwoeijf';
+      const wrapper = mount(KTextbox, {
+        propsData: {
+          label: 'test',
+          value: initialValue,
+        },
+      });
+      const value = wrapper.find('input').element.value;
+      expect(value).toBe(initialValue);
+    });
+    it(`invalidText is not displayed if showInvalidText is false`, () => {
+      const wrapper = mount(KTextbox, {
+        propsData: {
+          invalid: true,
+          showInvalidText: false,
+          invalidText: 'error!',
+        },
+      });
+      const errorTextField = wrapper.find('.ui-textbox-feedback-text');
+      expect(errorTextField.text()).not.toBe('error!');
+    });
+    it(`invalidText is displayed through error prop if invalid and showInvalidText are both true`, () => {
+      const wrapper = mount(KTextbox, {
+        propsData: {
+          invalid: true,
+          showInvalidText: true,
+          invalidText: 'error!',
+        },
+      });
+      const errorTextField = wrapper.find('.ui-textbox-feedback-text');
+      expect(errorTextField.text()).toBe('error!');
+    });
+    it(`text field is disabled when 'disabled' is true`, () => {
+      const wrapper = mount(KTextbox, {
+        propsData: {
+          disabled: true,
+        },
+      });
+      expect(wrapper.find('input').attributes('disabled')).toBe('disabled');
```

Comment:
I don't totally understand why this would be `'disabled'` and not a boolean, although the test is passing

Reply:
`disabled` is only boolean by virtue of its presence or absence - that's not always been supported, if I recall correctly, so it's often been conventional to set the value to `'disabled'`.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled

Reply:
TIL it has something to do with xhtml: https://stackoverflow.com/a/6531804

Reply:
Ahah - thanks for the canonical explanation!